ios-device-lib-stdio-handler.js

```diff
@@ -0,0 +1,106 @@
+"use strict";
+
+const EventEmitter = require("events");
+const path = require("path");
+const os = require("os");
+const spawn = require("child_process").spawn;
+
+const bufferpack = require("bufferpack");
+const Constants = require("./constants");
+
+const HeaderSize = 4;
+
+class IOSDeviceLibStdioHandler extends EventEmitter {
+	constructor() {
+		super();
+		this._unfinishedMessage = new Buffer(0);
+	}
+
+	startReadingData() {
+		this._chProc = spawn(path.join(__dirname, "bin", os.platform(), os.arch(), "ios-device-lib"));
+		this._chProc.stdout.on("data", (data) => {
+			this._unpackMessages(data);
+		});
+	}
+
+	writeData(data) {
+		this._chProc.stdin.write(data);
+	}
+
+	dispose(signal) {
+		this.removeAllListeners();
+		this._chProc.removeAllListeners();
+		this._chProc.kill(signal);
+	}
+
+	_distributeMessage(message) {
+		if (message.event) {
+			switch (message.event) {
+				case Constants.DeviceEventEnum.kDeviceFound:
+					this.emit(Constants.DeviceFoundEventName, message);
+					break;
+				case DeviceEventEnum.kDeviceLost:
+					this.emit(Constants.DeviceLostEventName, message);
+					break;
+				case DeviceEventEnum.kDeviceTrusted:
+					this.emit(Constants.DeviceLostEventName, message);
+					this.emit(Constants.DeviceFoundEventName, message);
+					break;
+			}
+		} else {
+			this.emit(Constants.DataEventName, message);
+		}
+	}
+
+	_unpackMessages(data) {
+		if (!this._unfinishedMessage.length && data.length >= HeaderSize) {
+			// Get the message length header.
+			const messageSizeBuffer = data.slice(0, HeaderSize);
+			const messageLength = bufferpack.unpack(">i", messageSizeBuffer)[0];
+
+			if (messageLength > data.length - HeaderSize) {
+				// Less than one message in the buffer.
+				// Store the unfinished message untill the next call of the function.
+				this._unfinishedMessage = data;
+			} else if (data.length - 4 > messageLength) {
+				// More than one message in the buffer.
+				const messageBuffer = this._getMessageFromBuffer(data, messageLength);
+
+				// Get the rest of the message here.
+				// Since our messages are not separated by whitespace or newlie
+				// we do not need to add somethig when we slice the original buffer.
+				const slicedBuffer = data.slice(messageBuffer.length + HeaderSize);
+				this._distributeMessage(this._parseMessageFromBuffer(messageBuffer));
+				this._unpackMessages(slicedBuffer);
+			} else {
+				// One message in the buffer.
+				const messageBuffer = this._getMessageFromBuffer(data, messageLength);
+				this._distributeMessage(this._parseMessageFromBuffer(messageBuffer));
+			}
+		} else if (this._unfinishedMessage.length && data.length >= HeaderSize) {
+			// Append the new data to the unfinished message and try to unpack again.
+			const concatenatedMessage = Buffer.concat([this._unfinishedMessage, data]);
+
+			// Clear the unfinished message buffer.
+			this._unfinishedMessage = new Buffer(0);
```

Comment:
`new Buffer` is deprecated
https://nodejs.org/api/buffer.html#buffer_new_buffer_size